src/storage/src/render/upsert.rs

```diff
@@ -584,16 +682,23 @@ fn process_pending_values_batch(
         if let Some(decoded_key) = key {
             let (decoded_key, decoded_value): (_, Result<_, DataflowError>) =
                 match (decoded_key, data.value) {
-                    (Err(key_decode_error), Some(_)) => {
-                        let err = DataflowError::from(EnvelopeError::Upsert(
-                            UpsertError::KeyDecode(key_decode_error.clone()),
-                        ));
-                        (Err(key_decode_error), Err(err))
```

Comment:
I find this code way more readable than the `and_then`. Do we really need to preserve the value error if there is a key error?

Reply:
I can get rid of the `and_then`, yes! I don't think we need to preserve the value error. And, in fact, we can only write down one of the errors in the persist shard, because it's schema is `Result<Row, DataflowError>`

Reply:
Yeah, in that case, the old code that just clones the key decode error seems like the most reasonable!